spec/presenters/mainstream_browse_page_presenter_spec.rb

```diff
@@ -92,15 +94,96 @@
         end
 
         it "includes the content_ids of linked topics sorted by title" do
-          expect(presented_data[:links]).to eq({
-            "related_topics" => [alpha.content_id, bravo.content_id],
-          })
+          expect(presented_data[:links]["related_topics"]).to eq([
+            alpha.content_id,
+            bravo.content_id,
+          ])
+        end
+
+        it "is valid against the schema", :schema_test => true do
+          expect(presented_data).to be_valid_against_schema('mainstream_browse_page')
+        end
+      end
+    end
+
+    describe "linking to related pages" do
+
+      let(:top_level_page_1) { create(:mainstream_browse_page) }
+      let(:top_level_page_2) { create(:mainstream_browse_page) }
+
+      let(:second_level_page_1) { create(:mainstream_browse_page) }
+      let(:second_level_page_2) { create(:mainstream_browse_page) }
+      let(:second_level_page_3) { create(:mainstream_browse_page) }
+
+      before :each do
+        # Top level pages must have children
+        top_level_page_1.children = [second_level_page_1, second_level_page_2]
```

Comment:
Instead of doing this, you could assign the parents as you create the second-level pages.  ie:
```ruby
let(:second_level_page_1) { create(:mainstream_browse_page, :parent => top_level_page_1) }
```

That way you'd avoid the need for this before block entirely.  If there are issues with the lazy loading of `let` blocks, you can use `let!` instead.